src/UCTSearch.cpp

```diff
@@ -663,6 +669,22 @@ void UCTSearch::increment_playouts() {
     m_playouts++;
 }
 
+#ifdef USE_OPENCL
+extern std::atomic<size_t> batch_stats[];
+#endif
+
+void UCTSearch::timer(Time start, int time_for_move) {
+    using namespace std::chrono_literals;
+    Time elapsed;
+    int elapsed_centis = Time::timediff_centis(start, elapsed);
+    while(!predict_stop_thinking(elapsed_centis, time_for_move)) {
+        Time elapsed2;
+        elapsed_centis = Time::timediff_centis(start, elapsed2);
+        std::this_thread::sleep_for(200ms);
+    }
+    m_network.set_batching(false);
+}
+
```

Comment:
It's actually a brilliant idea! But it requires all threads knows when to stop. Current implementation only main thread knows when to stop thinking, while other threads just pull is_running. So if main thread waits in the queue, it'll stuck.

I still like this idea, we could improve the code a bit to see if we can do this.

Reply:
The main thread could signal a bunch of notification vars, and this sleep_for could be a timed_wait for those.

Reply:
After some thoughts, it appears to me simply turning off batching after main thread exits the loop would be enough. Only thing needs to worry is when node children are not enough to form a batch, we can just turn off batching for this case since move choices are not that many anyway.